### PR TITLE
Remove QUERY_AND_FETCH BWC for pre-5.3.0 nodes

### PR DESCRIPTION
**Pull request by dakrone**
This was a BWC layer where we expicitly set the `search_type` to
"query_and_fetch" when a single node is queried on pre-5.3 nodes. Since 6.0 no
longer needs to be compatible with 5.3 nodes, this can be removed.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/action/search/SearchTransportService.java [hunk 1]
```diff
@@ -117,26 +117,11 @@ public void sendExecuteDfs(Transport.Connection connection, final ShardSearchTra
...
-            if (request.scroll() != null) {
-                /**
-                 * This is needed for nodes pre 5.3 when the single shard optimization is used.
-                 * These nodes will set the last emitted doc only if the removed `query_and_fetch` search type is set
-                 * in the request. See {@link SearchType}.
-                 */
-                request.searchType(SearchType.QUERY_AND_FETCH);
-            }
-            // TODO this BWC layer can be removed once this is back-ported to 5.3
-            transportService.sendChildRequest(connection, QUERY_FETCH_ACTION_NAME, request, task,
```
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/action/search/SearchTransportService.java]: does the `QUERY_FETCH_ACTION_NAME` constant still need to exist?

File: core/src/main/java/org/elasticsearch/action/search/SearchTransportService.java [hunk 2]
```diff
@@ -117,26 +117,11 @@ public void sendExecuteDfs(Transport.Connection connection, final ShardSearchTra
...
-                new ActionListenerResponseHandler<>(listener, supplier));
-        } else {
             transportService.sendChildRequest(connection, QUERY_ACTION_NAME, request, task,
                 new ActionListenerResponseHandler<>(listener, supplier));
```
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/action/search/SearchTransportService.java]: let's fix indentation?

**Review by jpountz — APPROVED** [review]
I left two minor comments. Thanks for removing dead code!

File: core/src/main/java/org/elasticsearch/action/search/SearchTransportService.java (hunk 1 above)
Comment (dakrone) [inline core/src/main/java/org/elasticsearch/action/search/SearchTransportService.java]: Nope! I removed those

File: core/src/main/java/org/elasticsearch/action/search/SearchTransportService.java (hunk 2 above)
Comment (dakrone) [inline core/src/main/java/org/elasticsearch/action/search/SearchTransportService.java]: Whoops yeah, I'll fix this.

**Review by s1monw — APPROVED** [review]
LGTM

**Comment by dakrone** [comment]
Thanks @s1monw and @jpountz!

**Pull request merged**